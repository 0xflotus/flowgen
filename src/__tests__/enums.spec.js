import { compiler, beautify } from "..";

it("should handle empty enums", () => {
  const ts = `enum Empty { }`;
  const result = compiler.compileDefinitionString(ts);
  expect(beautify(result)).toMatchSnapshot();
});

it("should handle basic enums", () => {
  const ts = `enum Label {
    LABEL_OPTIONAL,
    LABEL_REQUIRED,
    LABEL_REPEATED,
  }`;
  const result = compiler.compileDefinitionString(ts);
  expect(beautify(result)).toMatchSnapshot();
});

it("should handle number enums", () => {
  const ts = `enum Label {
    ONE = 1,
    TWO = 2,
    THREE = 3,
    NEGATIVE = -123,
    DECIMAL = 3.14,
  }`;
  const result = compiler.compileDefinitionString(ts);
  expect(beautify(result)).toMatchSnapshot();
});

it("should handle string enums", () => {
  const ts = `enum Label {
    LABEL_OPTIONAL = 'LABEL_OPTIONAL',
    LABEL_REQUIRED = 'LABEL_REQUIRED',
    LABEL_REPEATED = 'LABEL_REPEATED',
  }`;
  const result = compiler.compileDefinitionString(ts);
  expect(beautify(result)).toMatchSnapshot();
});
