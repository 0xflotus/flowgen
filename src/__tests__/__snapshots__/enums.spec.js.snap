// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle basic enums: class 1`] = `
"declare class Label {
  constructor(...args: empty): mixed;
  static +LABEL_OPTIONAL: Class<Label__LABEL_OPTIONAL> &
    Label__LABEL_OPTIONAL &
    0; // 0
  static +LABEL_REQUIRED: Class<Label__LABEL_REQUIRED> &
    Label__LABEL_REQUIRED &
    1; // 1
  static +LABEL_REPEATED: Class<Label__LABEL_REPEATED> &
    Label__LABEL_REPEATED &
    2; // 2
}

declare class Label__LABEL_OPTIONAL mixins Label {}
declare class Label__LABEL_REQUIRED mixins Label {}
declare class Label__LABEL_REPEATED mixins Label {}
"
`;

exports[`should handle basic enums: union 1`] = `
"declare type Label = \\"LABEL_OPTIONAL\\" | \\"LABEL_REQUIRED\\" | \\"LABEL_REPEATED\\";
"
`;

exports[`should handle empty enums: class 1`] = `
"declare class Empty {
  constructor(...args: empty): mixed;
}
"
`;

exports[`should handle empty enums: union 1`] = `
"declare type Empty = empty;
"
`;

exports[`should handle number enums: class 1`] = `
"declare class Label {
  constructor(...args: empty): mixed;
  static +ONE: Class<Label__ONE> & Label__ONE & 1; // 1
  static +TWO: Class<Label__TWO> & Label__TWO & 2; // 2
  static +THREE: Class<Label__THREE> & Label__THREE & 3; // 3
  static +NEGATIVE: Class<Label__NEGATIVE> & Label__NEGATIVE & -123; // -123
  static +DECIMAL: Class<Label__DECIMAL> & Label__DECIMAL & 3.14; // 3.14
}

declare class Label__ONE mixins Label {}
declare class Label__TWO mixins Label {}
declare class Label__THREE mixins Label {}
declare class Label__NEGATIVE mixins Label {}
declare class Label__DECIMAL mixins Label {}
"
`;

exports[`should handle number enums: union 1`] = `
"declare type Label = \\"ONE\\" | \\"TWO\\" | \\"THREE\\" | \\"NEGATIVE\\" | \\"DECIMAL\\";
"
`;

exports[`should handle string enums: class 1`] = `
"declare class Label {
  constructor(...args: empty): mixed;
  static +LABEL_OPTIONAL: Class<Label__LABEL_OPTIONAL> &
    Label__LABEL_OPTIONAL &
    \\"LABEL_OPTIONAL\\"; // \\"LABEL_OPTIONAL\\"
  static +LABEL_REQUIRED: Class<Label__LABEL_REQUIRED> &
    Label__LABEL_REQUIRED &
    \\"LABEL_REQUIRED\\"; // \\"LABEL_REQUIRED\\"
  static +LABEL_REPEATED: Class<Label__LABEL_REPEATED> &
    Label__LABEL_REPEATED &
    \\"LABEL_REPEATED\\"; // \\"LABEL_REPEATED\\"
}

declare class Label__LABEL_OPTIONAL mixins Label {}
declare class Label__LABEL_REQUIRED mixins Label {}
declare class Label__LABEL_REPEATED mixins Label {}
"
`;

exports[`should handle string enums: union 1`] = `
"declare type Label = \\"LABEL_OPTIONAL\\" | \\"LABEL_REQUIRED\\" | \\"LABEL_REPEATED\\";
"
`;
